File: smash-ranks-android/app/src/main/java/com/garpr/android/misc/GoogleApiWrapper.java
package com.garpr.android.misc;

import android.app.Activity;
import android.content.DialogInterface.OnCancelListener;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.google.android.gms.gcm.GcmNetworkManager;

public interface GoogleApiWrapper {

    GcmNetworkManager getGcmNetworkManager();

    int getGooglePlayServicesConnectionStatus();

    boolean isConnectionStatusSuccess(final int connectionStatus);

    boolean isGooglePlayServicesAvailable();

    boolean showPlayServicesResolutionDialog(final int connectionStatus,
            @NonNull final Activity activity, @Nullable final OnCancelListener onCancelListener);

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/misc/GoogleApiWrapper.kt
package com.garpr.android.misc

import android.app.Activity
import android.content.DialogInterface.OnCancelListener

interface GoogleApiWrapper {

    val googlePlayServicesConnectionStatus: Int

    fun isConnectionStatusSuccess(connectionStatus: Int): Boolean

    val isGooglePlayServicesAvailable: Boolean

    fun showPlayServicesResolutionDialog(activity: Activity, connectionStatus: Int,
            onCancelListener: OnCancelListener?): Boolean

}
